### PR TITLE
Prod Nomination 0

### PR DESCRIPTION
**Pull request by Sanjay-B**
Node Version has been bumped to **v16.17.1**. This version is **required**. 

## What's Changed
* Added Nextjs Framework, TailwindCSS & MaterialUi dependencies, wrote Readme, project setup as Web-based Application by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/1
* Implemented Protected Routes, Authentication (Oauth2), Session Tracking & Basic Error Page Handling by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/2
* Bump minimist from 1.2.5 to 1.2.6 in /src by @dependabot in https://github.com/KSU-Google-DSC/Webapp/pull/4
* Setup Working Project, Landing Page implementation, Navbar Implementation by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/18
* Fix merge conflicts. by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/21
* Bump next-auth from 4.2.1 to 4.9.0 in /src by @dependabot in https://github.com/KSU-Google-DSC/Webapp/pull/20
* Bump minimist from 1.2.5 to 1.2.6 in /src by @dependabot in https://github.com/KSU-Google-DSC/Webapp/pull/19
* Bump next-auth from 4.9.0 to 4.10.3 in /src by @dependabot in https://github.com/KSU-Google-DSC/Webapp/pull/26
* Improve Error Page Design & Better Descriptive Errors by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/30
* Implement Responsive Navbar (mobile) and Auto-rescaling Web Content by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/24
* Support Firebase Configuration in Production Env File by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/29
* Implement Wave Animation to Landing Page by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/27
* Implement About Page by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/33
* Implement Event Page by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/34
* Implement Team Page by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/35
* Implement Socials Page by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/36
* Implement Recruitment Page by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/39
* Bump jose from 4.8.3 to 4.9.3 in /src by @dependabot in https://github.com/KSU-Google-DSC/Webapp/pull/40
* Implement Account Profiles, Google Provider Login, Authentication Tracking by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/6
* Implement API endpoints for recruitment by @johnnymcmike in https://github.com/KSU-Google-DSC/Webapp/pull/25
* Add event endpoint (GET /api/events/all) by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/42
* Updating Link in Homepage by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/43
* Support GCP Deployment through Docker by @Sanjay-B in https://github.com/KSU-Google-DSC/Webapp/pull/45
* Add prettier as a pre-commit hook for easy file formatting by @cgodwin0 in https://github.com/KSU-Google-DSC/Webapp/pull/51
* Remove unauthenticated api/applicants/submit endpoint by @johnnymcmike in https://github.com/KSU-Google-DSC/Webapp/pull/57
* Fix Invalid Tab and Link components by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/55
* Bugfix/fix aboutpage by @Hoangnguy02 in https://github.com/KSU-Google-DSC/Webapp/pull/61
* New Github Action to run Cypress Front end Tests by @teamrocket77 in https://github.com/KSU-Google-DSC/Webapp/pull/62
* Fixed several errors that still shown in the console by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/69
* Convert all endpoints into a handler that calls an exported async function by @johnnymcmike in https://github.com/KSU-Google-DSC/Webapp/pull/68
* Implemented Login Page by @aimiisme in https://github.com/KSU-Google-DSC/Webapp/pull/63

**Full Changelog**: https://github.com/KSU-Google-DSC/Webapp/commits/v0.0.0

### CODE REVIEW

**Review by PremKSU — APPROVED** [review]
Most front-end changes look good to me. Hamburger menu, get started button, and navbar home button seem to be working fine. I cannot speak on the functionality of backend right now. Can we remove the logo from the About page? Can comment it out for now, and update it when we have a better picture to put in.

**Review by aimiisme — APPROVED** [review]
I approve those commits for now, even though we still need to develop this project. Those are ready to be merged into the `main` branch.

**Comment by Sanjay-B** [comment]
Appreciate the reviews thus far. I am going to hold off until we get deployment working. Blocked by https://github.com/KSU-Google-DSC/Webapp/pull/45 . @PremKSU cc: @aimiisme , @teamrocket77

**Comment by Sanjay-B** [comment]
Where are we with this? @PremKSU

**Comment by Sanjay-B** [comment]
**Blocked: ** RC0 (this PR) fails to build docker during deployment via Github Actions. I am looking into a fix. Est: ~After Tuesday (10/23/22).

**Comment by Sanjay-B** [comment]
**Blocked:** RC0 (this PR) fails to build docker during deployment via Github Actions. I am looking into a fix. Est: ~After Tuesday (10/23/22). 

cc: @PremKSU

**Comment by Sanjay-B** [comment]
Blocked: RC0 (this PR) fails to build docker during deployment via Github Actions. I am looking into a fix. Deployment keys are valid and are present as secrets in the repo. Est: ~idk

Still blocked by the issue. Can't seem to find a fix.

cc: @PremKSU

File: .github/workflows/gcr_deployment.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
+name: webapp-deployment-gcp
+on:
+  release:
```
Comment (teamrocket77) [inline .github/workflows/gcr_deployment.yml:3]: Do we have any successful runs for this action? if not there is a `workflow_dispatch` option
[Manually running](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow)
[workflow_dispatch event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+NEXTAUTH_SECRET: ""
+GOOGLE_PROVIDER_CLIENT_ID: ""
+GOOGLE_PROVIDER_CLIENT_SECRET: ""
+GOOGLE_PROVIDER_AUTHORIZATION_URL: ""
+FIREBASE_SERVICE_ACCOUNT_PROJECT_ID: ""
+FIREBASE_SERVICE_ACCOUNT_PRIVATE_KEY: ""
+FIREBASE_SERVICE_ACCOUNT_CLIENT_EMAIL: ""
+```
+
+`SECRET` and `NEXTAUTH_SECRET` can be created by typing `openssl rand -base64 32` into the terminal.
```
Comment (teamrocket77) [inline CONTRIBUTING.md:58]: We longer use pre generated creds @Sanjay-B ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+COPY /src/package.json ./
+COPY /src/package-lock.json ./
+RUN ls -a
+RUN npm ci --only=production
+
+# Build
+FROM node:16-alpine AS builder
+WORKDIR /app
+COPY . .
+COPY /src/.env /app/src
```
Comment (teamrocket77) [inline Dockerfile:16]: this should take care of all the copying
why are we copying .env & package*.json files file separately? 
Given the current nature of the project does this build for you?
@Sanjay-B

**Review by teamrocket77 — CHANGES_REQUESTED** [review]
Just looking for some answers, lots of things have changed

File: Dockerfile (hunk 3 above)
Comment (Sanjay-B) [reply Dockerfile:16]: Dependencies step installs only dependencies. @teamrocket77

File: CONTRIBUTING.md (hunk 2 above)
Comment (Sanjay-B) [reply CONTRIBUTING.md:58]: For dev environments, they'd still have to pre-generate this.

File: .github/workflows/gcr_deployment.yml (hunk 1 above)
Comment (Sanjay-B) [reply .github/workflows/gcr_deployment.yml:3]: Yeah, I think that option is good. I am going to add it in. @teamrocket77

**Comment by Sanjay-B** [comment]
**Update**: Deployment is being looked into by @Sanjay-B , @teamrocket77 . We've narrowed it down to build keys being passed via the Docker container and failing to be recognized by the build environment when the application enters into start-mode. Looking into possible solutions. RC0 will now be shipped with Cypress (Testing Environment). We're still looking into the issue that this poses for the Events page. I will follow up in internal channels.

**Comment by Sanjay-B** [comment]
Update: RC0 is no longer blocked. Produces runnable build & deployment.

**Comment by Sanjay-B** [comment]
Update: Blocked by missing functionality in Submit button in ``/opportunities`` and ``/team`` not being connected to the backend. 
cc: @aimiisme , @johnnymcmike

**Comment by Sanjay-B** [comment]
Pushed RC0 release to ``1/6/23`` to give time to fix said issues.

**Review by Sanjay-B — COMMENTED** [review]
LGTM. Merging into ``main``.

**Pull request merged**